### PR TITLE
Fix: #580 Create ModalCommandPalette

### PR DESCRIPTION
**Pull request by mattburnett-repo**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

This PR addresses Issue #580. It creates a baseline for the component as described in the original issue.

It is only tested on MacOS. 

There are some TODO tasks:
   │  │  └─ ModalCommandPalette.vue
   │  │     ├─ line 18: TODO : Input area is much smaller that tag width.
   │  │     ├─ line 37: TODO : Sort out router-link target routes (the to="..." things).
   │  │     └─ line 170: TODO : Maybe there is a better / more future-proof solution than userAgent?

### Related issue
- #580

### CODE REVIEW

**Comment by andrewtavis** [comment]
Thanks @mattburnett-repo! Do you want to check the errors reported above, or should I get to them?

**Comment by andrewtavis** [comment]
> Error:   19:13  error  The element inside `<transition>` is expected to have a `v-if` or `v-show` directive  vue/require-toggle-inside-transition

I'd be happy to do the i18n ones on review :)

**Comment by mattburnett-repo** [comment]
I made the errors. I'll fix them. Should have another commit / PR in maybe an hour.

**Comment by mattburnett-repo** [comment]
fixed the i18n errors in the ModalCommandPalette code.

the remaining i18n_check error involves code outside of the code for ModalCommandPalette. could you fix that one, please?

File: frontend/components/modal/ModalCommandPalette.vue [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+const { slash, meta_k, ctrl_k } = useMagicKeys({
+  passive: false,
+  onEventFired(e) {
+    if (["slash", "meta_k", "ctrl_k"].includes(e.key) && e.type === "keydown")
+      e.preventDefault();
+  },
+});
+
+whenever(slash, () => {
```
Comment (andrewtavis) [inline frontend/components/modal/ModalCommandPalette.vue:178]: Quick thing that I'm realizing here @mattburnett-repo is that I didn't do a good job of explaining that slash should maintain its old functionality for this :)

Ultimately activist will have two kinds of search: the slash one that's from the search bar that searches within the current domain for events or organizations or anything if we're not within a sub domain, and then the command palette that should only be triggered by `meta_k` or `ctrl_k`.

Would it be possible to:
- Remove the slash functionality from this
- Uncomment those lines in the search bar to show the command palette shortcut indicators

I'll then go through and do some styling. After that we should be good!

File: frontend/components/modal/ModalCommandPalette.vue [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+      </router-link>
+    </template>
+  </ModalBase>
+</template>
+
+<script setup>
+import { useMagicKeys, whenever } from "@vueuse/core";
+
+// TODO: Maybe there is a better / more future-proof solution than userAgent?
+const isMacOS = /Mac/.test(navigator.userAgent);
```
Comment (andrewtavis) [inline frontend/components/modal/ModalCommandPalette.vue:168]: You can use the device check module as is done [here](https://github.com/activist-org/activist/blob/cb57435f2fbb3e4262ca18e8b0404923d624526f/frontend/components/SearchBar.vue#L37C17-L37C17) 😊 These are also the lines we want to be uncommented :)

Comment (andrewtavis) [reply frontend/components/modal/ModalCommandPalette.vue:168]: This is as good of a solution as I could think of, and would be fine for now until we realize that it has major drawbacks.

File: frontend/components/modal/ModalCommandPalette.vue (hunk 1 above)
Comment (mattburnett-repo) [reply frontend/components/modal/ModalCommandPalette.vue:178]: No worries @andrewtavis . We had earlier some discussion about overall CommandPalette functionality and I probably got confused during that discussion. 

I've implemented your suggestions and they will appear in the next PR.

File: frontend/components/modal/ModalCommandPalette.vue (hunk 2 above)
Comment (mattburnett-repo) [reply frontend/components/modal/ModalCommandPalette.vue:168]: Got it. The $device.isMacOS (from the template example) doesn't work in a script block. It took me a minute, but I figured out to use useDevice().

Comment (andrewtavis) [reply frontend/components/modal/ModalCommandPalette.vue]: Awesome! Great to hear, @mattburnett-repo!

File: frontend/components/modal/ModalCommandPalette.vue (hunk 1 above)
Comment (andrewtavis) [reply frontend/components/modal/ModalCommandPalette.vue]: Looking forward to checking out the finalized version later, @mattburnett-repo! Thanks for the quick reaction :) :)

**Comment by andrewtavis** [comment]
Note to self, do not do Headwind formatting for the Tailwind classes while reviewing as it'll cause crazy merge conflicts 😅

**Review by andrewtavis — APPROVED** [review]
First version is done, @mattburnett-repo! This is really amazing stuff here 😊 There are still some minor things to do that I'll mention in the issue, but I think we're good to merge this in :)

**Pull request merged**